IPython/core/inputsplitter.py

```diff
@@ -698,46 +699,61 @@ def push(self, lines):
         # flush the buffer.
         self._store(lines, self._buffer_raw, 'source_raw')
 
+        transformed_lines_list = []
         for line in lines_list:
-            out = self.push_line(line)
+            transformed = self._transform_line(line)
+            if transformed is not None:
+                transformed_lines_list.append(transformed)
 
-        return out
-    
-    def push_line(self, line):
-        buf = self._buffer
+        if transformed_lines_list:
+            transformed_lines = '\n'.join(transformed_lines_list)
+            return super(IPythonInputSplitter, self).push(transformed_lines)
+        else:
+            # Got nothing back from transformers - they must be waiting for
+            # more input.
+            return False
+
+    def _transform_line(self, line):
+        """Push a line of input code through the various transformers.
+        
+        Returns any output from the transformers, or None if a transformer
+        is accumulating lines.
         
+        Sets self.transformer_accumulating as a side effect.
+        """
         def _accumulating(dbg):
             #print(dbg)
             self.transformer_accumulating = True
-            return False
-        
+            return None
+
         for transformer in self.physical_line_transforms:
             line = transformer.push(line)
             if line is None:
                 return _accumulating(transformer)
-        
+
         if not self.within_python_line:
             line = self.assemble_logical_lines.push(line)
             if line is None:
-                return _accumulating('acc logical line')        
-        
+                return _accumulating('acc logical line')
```

Comment:
@takluyver Is 'acc logical line' just representing that the line has been assembled?

Reply:
It indicates that the thing that swallowed this line without returning something is the logical line accumulator. These values are only used for debugging, which thankfully we haven't had to do very often.